app/mailers/notification_mailers/mentioned.rb

```diff
@@ -4,9 +4,10 @@ class Mentioned < NotificationMailers::Base
     delegate :author_name, to: :post, prefix: true
 
     def set_headers(target_id)
-      @post = Mention.find_by_id(target_id).post
+      @post = Mention.find_by_id(target_id).mentions_container
 
       @headers[:subject] = I18n.t('notifier.mentioned.subject', :name => @sender.name)
+      @headers[:template_name] = "mentioned_in_comment" if post.is_a?(Comment)
```

Comment:
I see what you are trying to do here ... and I'm unsure about it.

On one side you try to minimize code duplication, on the other side you need a workaround to make this work. Also you have `@post` here, which leads to stuff like `.post.parent` and `.post.guid` in the `mentioned_in_comment.markerb`.

If you create a separate class for `MentionedInComment` you would have a small code duplication, but you wouldn't need the workaround and you can name the variables properly (and you don't even need the `post_author_name` in the `mentioned_in_comment` template, so less duplication) and you don't need a `if` here.

And creating a separate class would also mean that you have separate checkboxes in the settings to enable mails for mentions on posts and comments.

At the end I'm more on the "split this completely"-side, I think that's the cleaner solution. What do you think?